common/unsafe/src/main/java/org/apache/spark/unsafe/array/ByteArrayMethods.java

```diff
@@ -40,6 +40,15 @@ public static int roundNumberOfBytesToNearestWord(int numBytes) {
     }
   }
 
+  public static long roundNumberOfBytesToNearestWord(long numBytes) {
+    long remainder = numBytes & 0x07;  // This is equivalent to `numBytes % 8`
```

Comment:
is it just `return (numBytes + 7) / 8;`?

Reply:
It's `8 * ((numBytes + 7) / 8)` but yeah, and the bit operation is unnecessarily complex IMHO. This is inherently a numeric operation, not bit-wise.

Reply:
Yes, it's 8 * ((numBytes + 7) / 8).